### PR TITLE
Few clean ups in ESBlobStoreRepositoryIntegTestCase

### PR DESCRIPTION
**Pull request by tlrx**
This pull request cleans up few things in ESBlobStoreRepositoryIntegTestCase. This is in preparation of removing mocks in S3BlobStoreRepositoryTests which will be done in a follow up PR.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java [hunk 1]
```diff
@@ -47,42 +48,44 @@
...
-
-            afterCreationCheck(repository);
-            assertThat("blob store has to be lazy initialized",
-                repository.getBlobStore(), verify ? is(notNullValue()) : is(nullValue()));
-        }
+        logger.debug("-->  creating repository [name: {}, verify: {}]", name, verify);
+        assertAcked(client().admin().cluster().preparePutRepository(name)
+            .setType(repositoryType())
+            .setVerify(verify)
+            .setSettings(Settings.builder().put(repositorySettings())));
```
Comment (original-brownbear) [inline test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java]: NIT: I don't think you need the builder here, there's a version of `setSettings` that just takes `Settings` ?

File: plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java [hunk 2]
```diff
@@ -71,21 +68,19 @@ public void wipeRepository() {
...
     @Override
-    protected void afterCreationCheck(Repository repository) {
-        assertThat(repository, instanceOf(S3Repository.class));
+    protected Settings repositorySettings() {
+        return Settings.builder()
+            .put(S3Repository.BUCKET_SETTING.getKey(), bucket)
+            .put(S3Repository.BUFFER_SIZE_SETTING.getKey(), bufferSize)
+            .put(S3Repository.SERVER_SIDE_ENCRYPTION_SETTING.getKey(), serverSideEncryption)
+            .put(S3Repository.CANNED_ACL_SETTING.getKey(), cannedACL)
+            .put(S3Repository.STORAGE_CLASS_SETTING.getKey(), storageClass)
```
Comment (original-brownbear) [inline plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java]: NIT: We seem to randomize `chunk_size` and `compress` for the Google and GS repository but not for the S3 one ... maybe just add that setting randomization to the abstract parent to dry things up and cover S3 as well?

**Review by original-brownbear — APPROVED** [review]
LGMT thanks @tlrx I left two optional suggestions :)

File: test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java (hunk 1 above)
Comment (tlrx) [reply test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java]: Right, thanks

File: plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java (hunk 2 above)
Comment (tlrx) [reply plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java]: Good point, I pushed 235ef53291e5b05608396a2880d8814d5210e240

File: test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java [hunk 3]
```diff
@@ -47,42 +50,50 @@
...
-    protected abstract void createTestRepository(String name, boolean verify);
-
-    protected void afterCreationCheck(Repository repository) {
+    protected abstract String repositoryType();
 
+    protected Settings repositorySettings() {
+        final Settings.Builder settings = Settings.builder();
+        settings.put("compress", randomBoolean());
+        if (randomBoolean()) {
+            long size = 1 << randomIntBetween(7, 10);
```
Comment (original-brownbear) [inline test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java]: This is too large :) GCS only allows up to 100Mb

Comment (tlrx) [reply test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java]: Yeah, I just saw the test failure :)

**Comment by tlrx** [comment]
Thanks @original-brownbear

**Pull request merged**